Add docs for mediaStreamTrackProcessor discardedFrames & totalFrames (#44770)
mdn/content: files/en-us/web/api/mediastreamtrackprocessor/totalframes/index.md

---
title: "MediaStreamTrackProcessor: totalFrames property"
short-title: totalFrames
slug: Web/API/MediaStreamTrackProcessor/totalFrames
page-type: web-api-instance-property
browser-compat: api.MediaStreamTrackProcessor.totalFrames
---

{{APIRef("Insertable Streams for MediaStreamTrack API")}}

> The **`totalFrames`** property of the {{domxref("MediaStreamTrackProcessor")}} interface returns a number indicating the total number of frames recieved by the processor.

Reviewer: ```suggestion
The **`totalFrames`** property of the {{domxref("MediaStreamTrackProcessor")}} interface returns a number indicating how many frames have been received by the processor in total.
```

## Value

A number.

> ## Examples

Reviewer: ```suggestion
## Examples

### Basic usage
```

All individual examples inside the "Examples" section need their own subheading, even if there is only one.

```js
async function init() {
  try {
    const stream = await navigator.mediaDevices.getUserMedia({ video: true });
    const [track] = stream.getVideoTracks();

    const processor = new MediaStreamTrackProcessor({
      track,
      maxBufferSize: 1,
    });
    const reader = processor.readable.getReader();

    while (true) {
      const { value: frame, done } = await reader.read();
      if (done) break;

>       // do something with frame...

Reviewer: ```suggestion
      // Do something with frame...
      
```
